src/main/scala/loamstream/drm/DrmChunkRunner.scala

```diff
@@ -78,11 +76,17 @@ final case class DrmChunkRunner(
     val resultsForSubChunks: Iterable[Observable[Map[LJob, RunData]]] = {
       import DrmTaskArray.fromCommandLineJobs
       
+      val maxJobsPerTaskArray = drmConfig.maxNumJobsPerTaskArray
+      
       for {
         (settings, rawJobs) <- subChunksBySettings(commandLineJobs)
-        drmTaskArray = fromCommandLineJobs(executionConfig, jobOracle, settings, drmConfig, pathBuilder, rawJobs)
+        rawJobChunk <- rawJobs.sliding(maxJobsPerTaskArray, maxJobsPerTaskArray)
       } yield {
-        runJobs(settings, drmTaskArray, shouldRestart)
+        val drmTaskArray = TimeUtils.time(s"Making DrmTaskArray with ${rawJobChunk.size} jobs", debug(_)) {
+          fromCommandLineJobs(executionConfig, jobOracle, settings, drmConfig, pathBuilder, rawJobChunk)
+        }
+        
+        runJobs(settings, drmTaskArray)
```

Comment:
Okay, so if I'm following the code, you used to run all jobs at once. Now you have a sliding window of jobs, run those, then move on to the next group. This works, but is sub-optimal (again, assuming I'm reading this right). Because it looks like this:

```
jobs = [0,1,2,3,4,5,6,7,8,9]
jobsWindow = [[0,1],[2,3],[4,5],[6,7],[8,9]]

for group in jobsWindow:
  execute(group)
  waitFor(group)
```

What sucks about this is that if job[0] takes 10 minutes and jobs[1..] might take 1 minute each. But, with the above, it takes 14 minutes to execute them all. If - instead - it was just a task queue of jobs and as one completes another can spin up immediately, the entire group will be done in 10 minutes.

I want to confirm my reading of this is correct. And, if so, you may want to update the way this is done to a job/task queue, and it should significantly reduce the time Ryan is waiting for LS jobs to run.

Reply:
> I want to confirm my reading of this is correct.

It isn't, fortunately.  It's more like: 
```
observablesForEachGroup = [execute(group) for group in jobsWindow]
...
observableOfResultsFromAllGroups = merge(observablesForEachGroup)
```
Or, `execute` (`runJobs` here) gives you an `Observable[A]`, which is an asynchronous stream of results.  There's no blocking operation analogous to `waitFor`.